### DISCUSSION TITLE
pythonfmu3 - Python Version info

### DISCUSSION THREAD

**Discussion by gricharuk** [community/community#125379] [Programming Help]
### Select Topic Area

Question

### Body

Hi, 

I've tried to use pythonfmu3, transferred a project pythonfmu, can anyone confirm:-
1. What version of python3 should be used - I'm need to use use python3.9 due to the platform
2. Using python3.11 - I can get it working, 
3. When I rebuild all using python3.9 - it looks like the .so files has a hard coded path to python3.11.4. When I instanciate it with fmpy I get that the python3.11 so file can't be found.
Thanks

5.

**Comment by CopperEagle** [comment]
Hey @gricharuk 

It is likely due to your environment setup. When looking at the repository, the CI actions test for Python 3.7, 3.8 and *3.9*. No mention of 3.11 in the first place. Also, the CI looks to be going just well. So make sure you use the right python version when installing it. To avoid some mistakes, try installing it in a virtual environment.


```bash
mkdir venv
cd venv

python39 -m venv .
source bin/activate
pip39 install pythonfmu3

python39 ../test.py
```

Hope this helps.

**Comment by mranv** [comment]
It sounds like you're encountering an issue with Python version compatibility when using pythonfmu3. PythonFMU is a Python package for creating Functional Mock-up Units (FMUs) for co-simulation with tools like FMPy.

When you're building the FMU with Python 3.9 and then trying to use it with FMPy, it seems like there's a discrepancy between the Python versions used during building and runtime. The error you're encountering suggests that the FMU is looking for Python 3.11, but you're trying to use it with Python 3.9.

Here are a few steps you can take to troubleshoot and potentially resolve this issue:

1. **Check PythonFMU3 Compatibility**: Ensure that PythonFMU3 is compatible with Python 3.9. Sometimes, newer versions of Python introduce changes that might not be backward compatible with older codebases.

2. **Build Environment**: Double-check your build environment to ensure that you're explicitly using Python 3.9 during the build process. This includes checking the environment variables, build scripts, or any configuration files related to PythonFMU3.

3. **Dynamic Linking**: It seems like there might be an issue with dynamic linking where the FMU is expecting specific Python libraries that are only available in Python 3.11. You might need to adjust the build settings to ensure that the FMU is linked against the correct Python version's libraries.

4. **Modify Build Scripts**: If PythonFMU3's build scripts are generating paths or configurations that are specific to Python 3.11, you may need to modify them to work with Python 3.9.

5. **Community Support**: Consider reaching out to the PythonFMU3 community or developers for further assistance. They may have encountered similar issues and could provide guidance or updates on compatibility with Python 3.9.

By investigating these areas, you should be able to identify and resolve the compatibility issue between PythonFMU3, Python 3.9, and FMPy. Let me know if you need further assistance!
